### ISSUE TITLE
Cadence sensor from wahoo only sporadically works

### ISSUE THREAD

**Issue by deadcyclo** [Sublimis/UrbanBiker#17]
I have an issue with a cadence sensor from wahoo (their speed sensor from the same pack has no issues). The issue is related to UrbanBiker, and not the sensor it self (I have no issues with the sensor with other apps, including int and bluetooth test apps). UrbanBiking only sporadically (perhaps once every 14 times) registers any data from the cadence sensor. The applies to both over ANT+ and bluetooth. Removing and adding the sensor from the app before the trip works sometimes, but not always. The sensor seems to work fine when used for speed instead of cadence. I think there potentially is an issue with the cadence implementation in the app itself?

Note: Ant software updated to latest version according to instructions in FAQ. But since the issue applies both to ANT and Bluetooth, it should make a difference anyway.

**Comment by Sublimis** [comment]
Thanks for the detailed explanation! There was a bug in versions before v4.22 (latest one, released a few days ago), that caused cadence sensor not to work if wheel circumference is not set, although (obviously) the wheel circumference is not needed for cadence. Please check that you are using v4.22 of the app. We cannot think of another reason why this wouldn't work as intended. Speed and cadence sensors use the exact same underlining algorithms and methods, with the only difference being the fact that one must also multiply by wheel circumference to get the speed. Could you please send us an example track (or at least cadence SVG graph) of the ride where the bug occured. We would also like to know where did the number 14 came from. This number could be a major clue, however unfortunately it tells us nothing at this time...

**Comment by deadcyclo** [comment]
Thanks for your prompt reply. I can confirm I'm on the latest version 4.22. 

Unfortunately I don't think the logs will help you at all, since this isn't sporadic. Either I get a reading at  the start of the trip, and I get cadence logs for the whole trip, or I don't get any readings for the trip at all, and no cadence graph whatsoever.

The number 14 isn't a clue I'm afraid, it's simply my best guess of opproximately how often I get the sensor to register. It can be two trips in a row, then nothing for several trips, and then a new trip that works.

Could the bug potentially still exist on v 4.22? I just ran down to the bicycle garage here at work and tried after setting a fake wheel circumference on the cadence sensor, and I immediately got a reading after starting a track and turning the crank. It might be coincidence, and it not be. I'll test it during my ride home, and rides tomorrow, and get back to you with my findings.

**Comment by Sublimis** [comment]
This appears to be our fault after all, sorry. The 'wheel circumference needed for cadence sensor' issue was only partially fixed: It was fixed for standalone cadence sensors, but combined speed&cadence sensors still suffer the issue in v4.22. We fixed this now, and can publish a fixed beta version if needed.

However, this may not explain the behavior where sensor works only on some rides and doesn't work on others. Have you deleted and re-added the sensor during that time? If yes, this should explain it (as wheel circumference gets lost when sensor gets deleted), and the issue should be resolved completely with the above fix.

Thanks once again for your feedback!

**Comment by deadcyclo** [comment]
Thanks for your feedback.

I can indeed confirm that after adding a random circumference the cadence sensor has worked for 6 trips.

To the second thing. The sensor was added and removed each time it started working. Once it stopped, usually after one ride, sometimes after two, it had to be removed and added again. Sometimes that would work, sometimes it didn't. Several times I also tried swapping speed/cadence back and forth when re-adding. I suspect somehow the automatic wheel circumference setting was what made it work sporadically after readding (but that's just a guess since I know nothing about the code).

Anyhow. Thanks for your help. And thanks for making a great product :+1: 
I'll go ahead and closet this issue, since things are working now.
